subsys/bluetooth/host/adv.c

```diff
@@ -51,11 +51,21 @@ struct ad_stream {
 	uint16_t current_ltv_offset;
 };
 
-static void ad_stream_new(struct ad_stream *stream, const struct bt_ad *ad, size_t ad_len)
+static size_t ad_stream_new(struct ad_stream *stream, const struct bt_ad *ad, size_t ad_len)
 {
+	size_t len = 0;
+
 	(void)memset(stream, 0, sizeof(*stream));
 	stream->ad = ad;
 	stream->ad_len = ad_len;
+
+	for (int i = 0; i < ad_len; i++) {
+		for (int j = 0; j < ad[i].len; j++) {
+			len += ad[i].data[j].data_len + 2;
```

Comment:
While it may be obvious to us (and right now), please add a comment stating that `+ 2` is from the LTV header (length and type).